### PR TITLE
Rewrite GitHub Actions workflows to use official Gradle actions

### PR DESCRIPTION
**Pull request by CommanderTvis**
1. Official Gradle actions are used.
2. Other actions are just updated.

### CODE REVIEW

**Review by nbirillo — COMMENTED** [review]
Thank you! I have some questions

File: .github/workflows/detekt.yml [hunk 1]
```diff
@@ -8,22 +8,16 @@ on:
...
+        uses: gradle/wrapper-validation-action@v1.0.4
+      - uses: gradle/gradle-build-action@v2.1.5
         with:
-          arguments: detektCheckAll
-          properties: |
-            org.gradle.caching=true
-          gradle-version: wrapper
-          # additional files to calculate key for dependency cache
-          gradle-dependencies-cache-key: |
-            buildSrc/**/Versions.kt
```
[the remark this answers is not in the document]
Comment (CommanderTvis) [reply .github/workflows/detekt.yml:29]: Gradle's action already does all the things related to caching with correct keys etc.

File: .github/workflows/diktat.yml [hunk 2]
```diff
@@ -8,22 +8,16 @@ on:
...
+        uses: gradle/wrapper-validation-action@v1.0.4
+      - uses: gradle/gradle-build-action@v2.1.5
         with:
-          arguments: diktatCheckAll
-          properties: |
-            org.gradle.caching=true
-          gradle-version: wrapper
-          # additional files to calculate key for dependency cache
-          gradle-dependencies-cache-key: |
-            buildSrc/**/Versions.kt
```
[the remark this answers is not in the document]
Comment (CommanderTvis) [reply .github/workflows/diktat.yml:29]: Same:
Gradle's action already does all the things related to caching with correct keys etc.

**Review by nbirillo — APPROVED** [review]
Great!

**Pull request merged**